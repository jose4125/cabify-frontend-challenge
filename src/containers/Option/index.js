import React from 'react';

export class Option extends React.Component {
  constructor(props) {
    super(props);
    this.OptionRefSelected = React.createRef();
  }

  componentDidMount() {
    if (this.props.counter === this.props.index) {
      this.props.handleOptionselectedElement(this.OptionRefSelected);
    }
  }
  componentDidUpdate(prevProps) {
    if (
      prevProps.counter !== this.props.index &&
      this.props.counter === this.props.index
    ) {
      this.props.handleOptionselectedElement(this.OptionRefSelected);
    }
  }
  render() {
    return (
      <li
        ref={this.OptionRefSelected}
        className={`dropdown__item ${
          this.props.countryCode.dial_code === this.props.countryCodeSelected
            ? 'selected'
            : ''
        }`}
        onClick={event => {
          this.props.setCountrySelected(this.props.countryCode.dial_code);
          this.props.handleOpenCloseDropdown(false);
        }}
      >
        <div className="dropdown__country">
          <span className={`flag ${this.props.countryCode.code}`} />
          <p className="dropdown__country-name">
            {this.props.countryCode.name}
          </p>
        </div>
        {this.props.countryCode.dial_code}
      </li>
    );
  }
}

export default Option;
