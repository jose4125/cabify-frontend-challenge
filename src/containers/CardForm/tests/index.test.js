import React from 'react';
import { shallow, mount } from 'enzyme';
import fetchMock from 'fetch-mock';

import CardForm from '../index';
import Option from '../../Option/index';
import mockData from '../../../../public/data/countryCodes.json';
import { wrap } from 'module';

describe('CardFrom', () => {
  const mockHandleChange = jest.fn();
  const mockHandlePrefix = jest.fn();
  const mockHandleReset = jest.fn();

  // handleChange,
  // handlePrefix,
  // handleReset

  let form = {
    fullName: {
      valid: false,
      touched: false,
      focus: false,
      dirty: false,
      validation: {
        required: true,
        onlyString: true
      }
    },
    jobDescription: {
      valid: false,
      touched: false,
      focus: false,
      dirty: false,
      validation: {
        required: true
      }
    },
    phoneNumber: {
      valid: false,
      touched: false,
      focus: false,
      dirty: false,
      validation: {
        required: true,
        minLength: 10,
        maxLength: 10,
        onlyNumbers: true
      }
    },
    email: {
      valid: false,
      touched: false,
      focus: false,
      dirty: false,
      validation: {
        required: true,
        email: true
      }
    },
    website: {
      touched: false,
      focus: false,
      dirty: false
    },
    address: {
      valid: false,
      touched: false,
      focus: false,
      dirty: false,
      validation: {
        required: true
      }
    },
    formIsValid: false
  };

  const context = {
    state: {
      fullName: '',
      jobDescription: '',
      prefix: '',
      phoneNumber: '',
      email: '',
      website: '',
      address: ''
    },
    handleReset: mockHandleReset
  };
  const wrapper = shallow(<CardForm context={context} />);

  it('should render with out creashing', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call handleReset when the form was submitted', () => {
    wrapper.find('.form').simulate('submit', { preventDefault() {} });
    expect(mockHandleReset).toHaveBeenCalledWith(
      wrapper.state().countryCodeSelected
    );
  });

  it('should the submit button be disable', () => {
    expect(wrapper.find('.button').prop('className')).toContain('disabled');
  });

  it('should the submit button be enable', () => {
    form.formIsValid = true;
    wrapper.setState({ form });
    expect(wrapper.find('.button').prop('className')).not.toContain('disabled');
  });

  it('should set the countryCodeSelected', () => {
    wrapper.instance().handleCountryCodeSelected('+39');
    expect(wrapper.state().countryCodeSelected).toBe('+39');
  });

  it('should toggle the isOpen value', () => {
    wrapper.instance().handleOpenCloseDropdown();
    expect(wrapper.state().isOpen).toBe(true);
    wrapper.instance().handleOpenCloseDropdown();
    expect(wrapper.state().isOpen).toBe(false);
  });

  it('should set the isOpen value to false', () => {
    wrapper.setState({ isOpen: true });
    wrapper.instance().handleOpenCloseDropdown(false);
    expect(wrapper.state().isOpen).toBe(false);
  });

  it('should set the counter value', () => {
    wrapper.setState({
      codes: [
        {
          name: 'Afghanistan',
          dial_code: '+93',
          code: 'AF'
        },
        {
          name: 'Albania',
          dial_code: '+355',
          code: 'AL'
        },
        {
          name: 'Algeria',
          dial_code: '+213',
          code: 'DZ'
        }
      ]
    });
    wrapper.instance().setIndexSelected('+355');
    expect(wrapper.state().counter).toBe(1);
  });

  it('should close the select if handleCloseClickOutside is true', () => {
    wrapper.instance().handleClickOutside();
    expect(wrapper.state().isOpen).toBe(false);
  });

  it('should keep open the select if handleCloseClickOutside is false', () => {
    wrapper.setState({ handleCloseClickOutside: false });
    wrapper.instance().handleClickOutside();
    expect(wrapper.state().isOpen).toBe(false);
  });
});

describe('CardFrom Mounted', () => {
  const context = {
    state: {
      fullName: '',
      jobDescription: '',
      prefix: '',
      phoneNumber: '',
      email: '',
      website: '',
      address: ''
    }
  };
  const wrapper = shallow(<CardForm context={context} />);

  it('should get the data when is mounted', async () => {
    fetchMock.mock('/data/countryCodes.json', mockData);
    const data = await fetch('/data/countryCodes.json');
    const response = await data.json();
    expect(response).toEqual(mockData);
    wrapper.setState({ codes: response.codes });
    expect(wrapper.state().codes).toEqual(response.codes);
  });
});
