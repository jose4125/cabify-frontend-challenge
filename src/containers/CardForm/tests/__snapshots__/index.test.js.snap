// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardFrom should render with out creashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardForm
    context={
      Object {
        "handleReset": [MockFunction],
        "state": Object {
          "address": "",
          "email": "",
          "fullName": "",
          "jobDescription": "",
          "phoneNumber": "",
          "prefix": "",
          "website": "",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        className="form"
        onSubmit={[Function]}
      >
        <div
          className="row"
        >
          <FormField
            isDirty={false}
            isFocus={false}
            isValid={false}
            label="Full name"
            name="fullname"
            onHandleBlur={[Function]}
            onHandleChange={[Function]}
            onHandleFocus={[Function]}
            stateName="fullName"
            touched={false}
            type="text"
            value=""
          />
        </div>
        <div
          className="row row-separationMedium"
        >
          <FormField
            isDirty={false}
            isFocus={false}
            isValid={false}
            label="Job description"
            name="jobdescription"
            onHandleBlur={[Function]}
            onHandleChange={[Function]}
            onHandleFocus={[Function]}
            stateName="jobDescription"
            touched={false}
            type="text"
            value=""
          />
        </div>
        <div
          className="row row-separationMedium row-gutterMedium"
        >
          <div
            className="col col3"
          />
          <FormField
            col="9"
            isDirty={false}
            isFocus={false}
            isValid={false}
            label="Phone number"
            name="phonenumber"
            onHandleBlur={[Function]}
            onHandleChange={[Function]}
            onHandleFocus={[Function]}
            stateName="phoneNumber"
            touched={false}
            type="tel"
            value=""
          />
        </div>
        <div
          className="row row-separationMedium"
        >
          <FormField
            isDirty={false}
            isFocus={false}
            isValid={false}
            label="Email"
            name="email"
            onHandleBlur={[Function]}
            onHandleChange={[Function]}
            onHandleFocus={[Function]}
            stateName="email"
            touched={false}
            type="email"
            value=""
          />
        </div>
        <div
          className="row row-separationMedium"
        >
          <FormField
            isDirty={false}
            isFocus={false}
            label="Website"
            name="website"
            onHandleBlur={[Function]}
            onHandleChange={[Function]}
            onHandleFocus={[Function]}
            stateName="website"
            touched={false}
            type="text"
            value=""
          />
        </div>
        <div
          className="row row-separationMedium"
        >
          <FormField
            isDirty={false}
            isFocus={false}
            isValid={false}
            label="Address"
            name="address"
            onHandleBlur={[Function]}
            onHandleChange={[Function]}
            onHandleFocus={[Function]}
            stateName="address"
            touched={false}
            type="text"
            value=""
          />
        </div>
        <div
          className="row row-separationHuge"
        >
          <input
            className="button button-full button-primary disabled"
            type="submit"
            value="Request"
          />
        </div>
      </form>,
      "className": "builder col col6",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="row"
          >
            <FormField
              isDirty={false}
              isFocus={false}
              isValid={false}
              label="Full name"
              name="fullname"
              onHandleBlur={[Function]}
              onHandleChange={[Function]}
              onHandleFocus={[Function]}
              stateName="fullName"
              touched={false}
              type="text"
              value=""
            />
          </div>,
          <div
            className="row row-separationMedium"
          >
            <FormField
              isDirty={false}
              isFocus={false}
              isValid={false}
              label="Job description"
              name="jobdescription"
              onHandleBlur={[Function]}
              onHandleChange={[Function]}
              onHandleFocus={[Function]}
              stateName="jobDescription"
              touched={false}
              type="text"
              value=""
            />
          </div>,
          <div
            className="row row-separationMedium row-gutterMedium"
          >
            <div
              className="col col3"
            />
            <FormField
              col="9"
              isDirty={false}
              isFocus={false}
              isValid={false}
              label="Phone number"
              name="phonenumber"
              onHandleBlur={[Function]}
              onHandleChange={[Function]}
              onHandleFocus={[Function]}
              stateName="phoneNumber"
              touched={false}
              type="tel"
              value=""
            />
          </div>,
          <div
            className="row row-separationMedium"
          >
            <FormField
              isDirty={false}
              isFocus={false}
              isValid={false}
              label="Email"
              name="email"
              onHandleBlur={[Function]}
              onHandleChange={[Function]}
              onHandleFocus={[Function]}
              stateName="email"
              touched={false}
              type="email"
              value=""
            />
          </div>,
          <div
            className="row row-separationMedium"
          >
            <FormField
              isDirty={false}
              isFocus={false}
              label="Website"
              name="website"
              onHandleBlur={[Function]}
              onHandleChange={[Function]}
              onHandleFocus={[Function]}
              stateName="website"
              touched={false}
              type="text"
              value=""
            />
          </div>,
          <div
            className="row row-separationMedium"
          >
            <FormField
              isDirty={false}
              isFocus={false}
              isValid={false}
              label="Address"
              name="address"
              onHandleBlur={[Function]}
              onHandleChange={[Function]}
              onHandleFocus={[Function]}
              stateName="address"
              touched={false}
              type="text"
              value=""
            />
          </div>,
          <div
            className="row row-separationHuge"
          >
            <input
              className="button button-full button-primary disabled"
              type="submit"
              value="Request"
            />
          </div>,
        ],
        "className": "form",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <FormField
              isDirty={false}
              isFocus={false}
              isValid={false}
              label="Full name"
              name="fullname"
              onHandleBlur={[Function]}
              onHandleChange={[Function]}
              onHandleFocus={[Function]}
              stateName="fullName"
              touched={false}
              type="text"
              value=""
            />,
            "className": "row",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "isDirty": false,
              "isFocus": false,
              "isValid": false,
              "label": "Full name",
              "name": "fullname",
              "onHandleBlur": [Function],
              "onHandleChange": [Function],
              "onHandleFocus": [Function],
              "stateName": "fullName",
              "touched": false,
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <FormField
              isDirty={false}
              isFocus={false}
              isValid={false}
              label="Job description"
              name="jobdescription"
              onHandleBlur={[Function]}
              onHandleChange={[Function]}
              onHandleFocus={[Function]}
              stateName="jobDescription"
              touched={false}
              type="text"
              value=""
            />,
            "className": "row row-separationMedium",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "isDirty": false,
              "isFocus": false,
              "isValid": false,
              "label": "Job description",
              "name": "jobdescription",
              "onHandleBlur": [Function],
              "onHandleChange": [Function],
              "onHandleFocus": [Function],
              "stateName": "jobDescription",
              "touched": false,
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="col col3"
              />,
              <FormField
                col="9"
                isDirty={false}
                isFocus={false}
                isValid={false}
                label="Phone number"
                name="phonenumber"
                onHandleBlur={[Function]}
                onHandleChange={[Function]}
                onHandleFocus={[Function]}
                stateName="phoneNumber"
                touched={false}
                type="tel"
                value=""
              />,
            ],
            "className": "row row-separationMedium row-gutterMedium",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": false,
                "className": "col col3",
              },
              "ref": null,
              "rendered": false,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "col": "9",
                "isDirty": false,
                "isFocus": false,
                "isValid": false,
                "label": "Phone number",
                "name": "phonenumber",
                "onHandleBlur": [Function],
                "onHandleChange": [Function],
                "onHandleFocus": [Function],
                "stateName": "phoneNumber",
                "touched": false,
                "type": "tel",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <FormField
              isDirty={false}
              isFocus={false}
              isValid={false}
              label="Email"
              name="email"
              onHandleBlur={[Function]}
              onHandleChange={[Function]}
              onHandleFocus={[Function]}
              stateName="email"
              touched={false}
              type="email"
              value=""
            />,
            "className": "row row-separationMedium",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "isDirty": false,
              "isFocus": false,
              "isValid": false,
              "label": "Email",
              "name": "email",
              "onHandleBlur": [Function],
              "onHandleChange": [Function],
              "onHandleFocus": [Function],
              "stateName": "email",
              "touched": false,
              "type": "email",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <FormField
              isDirty={false}
              isFocus={false}
              label="Website"
              name="website"
              onHandleBlur={[Function]}
              onHandleChange={[Function]}
              onHandleFocus={[Function]}
              stateName="website"
              touched={false}
              type="text"
              value=""
            />,
            "className": "row row-separationMedium",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "isDirty": false,
              "isFocus": false,
              "label": "Website",
              "name": "website",
              "onHandleBlur": [Function],
              "onHandleChange": [Function],
              "onHandleFocus": [Function],
              "stateName": "website",
              "touched": false,
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <FormField
              isDirty={false}
              isFocus={false}
              isValid={false}
              label="Address"
              name="address"
              onHandleBlur={[Function]}
              onHandleChange={[Function]}
              onHandleFocus={[Function]}
              stateName="address"
              touched={false}
              type="text"
              value=""
            />,
            "className": "row row-separationMedium",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "isDirty": false,
              "isFocus": false,
              "isValid": false,
              "label": "Address",
              "name": "address",
              "onHandleBlur": [Function],
              "onHandleChange": [Function],
              "onHandleFocus": [Function],
              "stateName": "address",
              "touched": false,
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <input
              className="button button-full button-primary disabled"
              type="submit"
              value="Request"
            />,
            "className": "row row-separationHuge",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "button button-full button-primary disabled",
              "type": "submit",
              "value": "Request",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "div",
        },
      ],
      "type": "form",
    },
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          className="form"
          onSubmit={[Function]}
        >
          <div
            className="row"
          >
            <FormField
              isDirty={false}
              isFocus={false}
              isValid={false}
              label="Full name"
              name="fullname"
              onHandleBlur={[Function]}
              onHandleChange={[Function]}
              onHandleFocus={[Function]}
              stateName="fullName"
              touched={false}
              type="text"
              value=""
            />
          </div>
          <div
            className="row row-separationMedium"
          >
            <FormField
              isDirty={false}
              isFocus={false}
              isValid={false}
              label="Job description"
              name="jobdescription"
              onHandleBlur={[Function]}
              onHandleChange={[Function]}
              onHandleFocus={[Function]}
              stateName="jobDescription"
              touched={false}
              type="text"
              value=""
            />
          </div>
          <div
            className="row row-separationMedium row-gutterMedium"
          >
            <div
              className="col col3"
            />
            <FormField
              col="9"
              isDirty={false}
              isFocus={false}
              isValid={false}
              label="Phone number"
              name="phonenumber"
              onHandleBlur={[Function]}
              onHandleChange={[Function]}
              onHandleFocus={[Function]}
              stateName="phoneNumber"
              touched={false}
              type="tel"
              value=""
            />
          </div>
          <div
            className="row row-separationMedium"
          >
            <FormField
              isDirty={false}
              isFocus={false}
              isValid={false}
              label="Email"
              name="email"
              onHandleBlur={[Function]}
              onHandleChange={[Function]}
              onHandleFocus={[Function]}
              stateName="email"
              touched={false}
              type="email"
              value=""
            />
          </div>
          <div
            className="row row-separationMedium"
          >
            <FormField
              isDirty={false}
              isFocus={false}
              label="Website"
              name="website"
              onHandleBlur={[Function]}
              onHandleChange={[Function]}
              onHandleFocus={[Function]}
              stateName="website"
              touched={false}
              type="text"
              value=""
            />
          </div>
          <div
            className="row row-separationMedium"
          >
            <FormField
              isDirty={false}
              isFocus={false}
              isValid={false}
              label="Address"
              name="address"
              onHandleBlur={[Function]}
              onHandleChange={[Function]}
              onHandleFocus={[Function]}
              stateName="address"
              touched={false}
              type="text"
              value=""
            />
          </div>
          <div
            className="row row-separationHuge"
          >
            <input
              className="button button-full button-primary disabled"
              type="submit"
              value="Request"
            />
          </div>
        </form>,
        "className": "builder col col6",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="row"
            >
              <FormField
                isDirty={false}
                isFocus={false}
                isValid={false}
                label="Full name"
                name="fullname"
                onHandleBlur={[Function]}
                onHandleChange={[Function]}
                onHandleFocus={[Function]}
                stateName="fullName"
                touched={false}
                type="text"
                value=""
              />
            </div>,
            <div
              className="row row-separationMedium"
            >
              <FormField
                isDirty={false}
                isFocus={false}
                isValid={false}
                label="Job description"
                name="jobdescription"
                onHandleBlur={[Function]}
                onHandleChange={[Function]}
                onHandleFocus={[Function]}
                stateName="jobDescription"
                touched={false}
                type="text"
                value=""
              />
            </div>,
            <div
              className="row row-separationMedium row-gutterMedium"
            >
              <div
                className="col col3"
              />
              <FormField
                col="9"
                isDirty={false}
                isFocus={false}
                isValid={false}
                label="Phone number"
                name="phonenumber"
                onHandleBlur={[Function]}
                onHandleChange={[Function]}
                onHandleFocus={[Function]}
                stateName="phoneNumber"
                touched={false}
                type="tel"
                value=""
              />
            </div>,
            <div
              className="row row-separationMedium"
            >
              <FormField
                isDirty={false}
                isFocus={false}
                isValid={false}
                label="Email"
                name="email"
                onHandleBlur={[Function]}
                onHandleChange={[Function]}
                onHandleFocus={[Function]}
                stateName="email"
                touched={false}
                type="email"
                value=""
              />
            </div>,
            <div
              className="row row-separationMedium"
            >
              <FormField
                isDirty={false}
                isFocus={false}
                label="Website"
                name="website"
                onHandleBlur={[Function]}
                onHandleChange={[Function]}
                onHandleFocus={[Function]}
                stateName="website"
                touched={false}
                type="text"
                value=""
              />
            </div>,
            <div
              className="row row-separationMedium"
            >
              <FormField
                isDirty={false}
                isFocus={false}
                isValid={false}
                label="Address"
                name="address"
                onHandleBlur={[Function]}
                onHandleChange={[Function]}
                onHandleFocus={[Function]}
                stateName="address"
                touched={false}
                type="text"
                value=""
              />
            </div>,
            <div
              className="row row-separationHuge"
            >
              <input
                className="button button-full button-primary disabled"
                type="submit"
                value="Request"
              />
            </div>,
          ],
          "className": "form",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <FormField
                isDirty={false}
                isFocus={false}
                isValid={false}
                label="Full name"
                name="fullname"
                onHandleBlur={[Function]}
                onHandleChange={[Function]}
                onHandleFocus={[Function]}
                stateName="fullName"
                touched={false}
                type="text"
                value=""
              />,
              "className": "row",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "isDirty": false,
                "isFocus": false,
                "isValid": false,
                "label": "Full name",
                "name": "fullname",
                "onHandleBlur": [Function],
                "onHandleChange": [Function],
                "onHandleFocus": [Function],
                "stateName": "fullName",
                "touched": false,
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <FormField
                isDirty={false}
                isFocus={false}
                isValid={false}
                label="Job description"
                name="jobdescription"
                onHandleBlur={[Function]}
                onHandleChange={[Function]}
                onHandleFocus={[Function]}
                stateName="jobDescription"
                touched={false}
                type="text"
                value=""
              />,
              "className": "row row-separationMedium",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "isDirty": false,
                "isFocus": false,
                "isValid": false,
                "label": "Job description",
                "name": "jobdescription",
                "onHandleBlur": [Function],
                "onHandleChange": [Function],
                "onHandleFocus": [Function],
                "stateName": "jobDescription",
                "touched": false,
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="col col3"
                />,
                <FormField
                  col="9"
                  isDirty={false}
                  isFocus={false}
                  isValid={false}
                  label="Phone number"
                  name="phonenumber"
                  onHandleBlur={[Function]}
                  onHandleChange={[Function]}
                  onHandleFocus={[Function]}
                  stateName="phoneNumber"
                  touched={false}
                  type="tel"
                  value=""
                />,
              ],
              "className": "row row-separationMedium row-gutterMedium",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": false,
                  "className": "col col3",
                },
                "ref": null,
                "rendered": false,
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "col": "9",
                  "isDirty": false,
                  "isFocus": false,
                  "isValid": false,
                  "label": "Phone number",
                  "name": "phonenumber",
                  "onHandleBlur": [Function],
                  "onHandleChange": [Function],
                  "onHandleFocus": [Function],
                  "stateName": "phoneNumber",
                  "touched": false,
                  "type": "tel",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <FormField
                isDirty={false}
                isFocus={false}
                isValid={false}
                label="Email"
                name="email"
                onHandleBlur={[Function]}
                onHandleChange={[Function]}
                onHandleFocus={[Function]}
                stateName="email"
                touched={false}
                type="email"
                value=""
              />,
              "className": "row row-separationMedium",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "isDirty": false,
                "isFocus": false,
                "isValid": false,
                "label": "Email",
                "name": "email",
                "onHandleBlur": [Function],
                "onHandleChange": [Function],
                "onHandleFocus": [Function],
                "stateName": "email",
                "touched": false,
                "type": "email",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <FormField
                isDirty={false}
                isFocus={false}
                label="Website"
                name="website"
                onHandleBlur={[Function]}
                onHandleChange={[Function]}
                onHandleFocus={[Function]}
                stateName="website"
                touched={false}
                type="text"
                value=""
              />,
              "className": "row row-separationMedium",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "isDirty": false,
                "isFocus": false,
                "label": "Website",
                "name": "website",
                "onHandleBlur": [Function],
                "onHandleChange": [Function],
                "onHandleFocus": [Function],
                "stateName": "website",
                "touched": false,
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <FormField
                isDirty={false}
                isFocus={false}
                isValid={false}
                label="Address"
                name="address"
                onHandleBlur={[Function]}
                onHandleChange={[Function]}
                onHandleFocus={[Function]}
                stateName="address"
                touched={false}
                type="text"
                value=""
              />,
              "className": "row row-separationMedium",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "isDirty": false,
                "isFocus": false,
                "isValid": false,
                "label": "Address",
                "name": "address",
                "onHandleBlur": [Function],
                "onHandleChange": [Function],
                "onHandleFocus": [Function],
                "stateName": "address",
                "touched": false,
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <input
                className="button button-full button-primary disabled"
                type="submit"
                value="Request"
              />,
              "className": "row row-separationHuge",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "button button-full button-primary disabled",
                "type": "submit",
                "value": "Request",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            "type": "div",
          },
        ],
        "type": "form",
      },
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": false,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
